### PR TITLE
Feature/refactors

### LINKED ISSUE #31: refactor the code to look more python-like
http://as.ynchrono.us/2007/12/filesystem-structure-of-python-project_21.html?m=1

### PR DESCRIPTION
**Pull request by neriat**

### CODE REVIEW

File: teitelbot/response/action.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from enum import Enum
+
+
+class Action(Enum):
+    NONE = 0
+    EMAIL_ACTION = 1
+    HAVING_ACTION = 2
+    REPEAT_SARCASTICALLY_ACTION = 3
+    DEFAULT_ACTION = 5
```
Comment (weissgilad) [inline teitelbot/response/action.py]: Is there supposed to be 4?

File: teitelbot/response/quote/__init__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from response import classifier
+from response.action import Action
+from response.exception import ResponseNotExistForAction
+
+LOGGER = logging.getLogger(__name__)
+
+
+def on_message(text: str, reply: Callable[[str], Any]) -> None:
+    actions = classifier.possible_classifications(text)
+    for action in actions:
```
Comment (weissgilad) [inline teitelbot/response/quote/__init__.py:13]: If there are several possible classifications, will teitelbot just spam a bunch of responses?

File: teitelbot/response/reply/__init__.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    is_empty = not full_name or msg == "מייל של"
+    return "של מי" if is_empty else email_generator.generate(full_name)
+
+
+def _generic_question_response() -> str:
+    return generic_response_generator.random_answer()
+
+
+def _repeat_word_sarcastically_response(text: str) -> str:
+    # for now (2021) sarcastically just means surround with quotes
```
Comment (weissgilad) [inline teitelbot/response/reply/__init__.py:102]: Lol

File: teitelbot/response/reply/__init__.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    significant_words = list(set(text.split()) & set(words_to_repeat_sarcastically))
```
Comment (weissgilad) [inline teitelbot/response/reply/__init__.py:103]: Python blatantly copying cpp

File: teitelbot/response/reply/service/email_generator.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+from util.string_utils import translate_to_heb, clean_punctuation
+
+
+def generate(full_name_in_hebrew: str) -> str:
+    name = _get_translated_name(full_name_in_hebrew)
+    suffix = '@g.jct.ac.il'
```
Comment (weissgilad) [inline teitelbot/response/reply/service/email_generator.py:6]: We should consider changing this

File: teitelbot/response/classifier.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+from constants.significant_words import words_to_repeat_sarcastically
+from response.action import Action
+from util import string_utils
+
+
+def classify(text: str) -> Action:
+    return possible_classifications(text)[0]
+
+
+def possible_classifications(text: str) -> List[Action]:
```
Comment (weissgilad) [inline teitelbot/response/classifier.py]: This function returns possible actions, not possible classifications

File: teitelbot/response/quote/__init__.py (hunk 2 above)
Comment (weissgilad) [reply teitelbot/response/quote/__init__.py:13]: The break is what stops it?
Is it normal in python to throw an exception even if the flow is good?

File: teitelbot/util/string_utils.py [hunk 7]
```diff
@@ -35,3 +35,7 @@ def replace_words(words: list, replacement_map: dict) -> list:
 
 def translate_to_heb(string_in_english: str) -> str:
     return translator.translate(string_in_english, src='iw', dest='en').text
+
+
+def is_true(string: str) -> bool:
```
Comment (weissgilad) [inline teitelbot/util/string_utils.py:40]: Ew...

File: teitelbot/client/telegram_client.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    def block_loop(self):
+        self.updater.idle()
+
+    def _start_web_hook(self):
+        self.updater.start_webhook(listen="0.0.0.0",
+                                   port=config.PORT,
+                                   url_path=config.TOKEN)
+        self.updater.bot.setWebhook(
+            f"https://{config.APP_NAME}.herokuapp.com/{config.TOKEN}")
```
Comment (weissgilad) [inline teitelbot/client/telegram_client.py:35]: Is the webhook always heroku?

File: teitelbot/response/classifier.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+from random import random
+from typing import List
+
+import config
+from constants.significant_words import words_to_repeat_sarcastically
+from response.action import Action
+from util import string_utils
+
+
+def classify(text: str) -> Action:
```
Comment (weissgilad) [inline teitelbot/response/classifier.py]: Unused

File: teitelbot/response/classifier.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    return possible_classifications(text)[0]
+
+
+def possible_classifications(text: str) -> List[Action]:
+    classifier_to_action = {
+        _check_number: Action.CHECK_NUMBER_ACTION,
```
Comment (weissgilad) [inline teitelbot/response/classifier.py:13]: Indent to make pretty

Comment (weissgilad) [reply teitelbot/response/classifier.py:13]: Or is that impossible because python sucks?

**Review by weissgilad — CHANGES_REQUESTED** [review]

File: teitelbot/response/action.py (hunk 1 above)
Comment (neriat) [reply teitelbot/response/action.py]: my bad.

File: teitelbot/response/quote/__init__.py (hunk 2 above)
Comment (neriat) [reply teitelbot/response/quote/__init__.py:13]: 1. no. only the first response.
2. idk

File: teitelbot/response/reply/service/email_generator.py (hunk 5 above)
Comment (neriat) [reply teitelbot/response/reply/service/email_generator.py:6]: to what suffix?
should we pass it passed via config?

File: teitelbot/util/string_utils.py (hunk 7 above)
Comment (neriat) [reply teitelbot/util/string_utils.py:40]: what do you suggest?

File: teitelbot/response/classifier.py (hunk 9 above)
Comment (neriat) [reply teitelbot/response/classifier.py]: i'll remove it

File: teitelbot/response/classifier.py (hunk 10 above)
Comment (neriat) [reply teitelbot/response/classifier.py:13]: image

File: teitelbot/response/reply/service/email_generator.py (hunk 5 above)
Comment (weissgilad) [reply teitelbot/response/reply/service/email_generator.py:6]: We should discuss this fully at the next weekly

File: teitelbot/util/string_utils.py (hunk 7 above)
Comment (weissgilad) [reply teitelbot/util/string_utils.py:40]: This is probably fine, just seems strange.
Is it really necessary? Is it some kind of convention?

File: teitelbot/response/action.py (hunk 1 above)
Comment (neriat) [reply teitelbot/response/action.py]: what about:
image

Comment (weissgilad) [reply teitelbot/response/action.py]: That might be easier to maintain, but choose whichever is more conventional

Comment (neriat) [reply teitelbot/response/action.py]: sorry couldn't make it format code

Comment (weissgilad) [reply teitelbot/response/action.py]: Meh

Comment (weissgilad) [reply teitelbot/response/action.py]: Is = auto() an option?

File: teitelbot/client/telegram_client.py (hunk 8 above)
Comment (neriat) [reply teitelbot/client/telegram_client.py:35]: unless we'll change our PAAS

File: teitelbot/util/string_utils.py (hunk 7 above)
Comment (neriat) [reply teitelbot/util/string_utils.py:40]: used in config parsing.
export a=false
considered as true boolean.

File: teitelbot/response/action.py (hunk 1 above)
Comment (neriat) [reply teitelbot/response/action.py]: what do u mean?

File: teitelbot/util/string_utils.py (hunk 7 above)
Comment (weissgilad) [reply teitelbot/util/string_utils.py:40]: Aight

File: teitelbot/response/action.py (hunk 1 above)
Comment (weissgilad) [reply teitelbot/response/action.py]: https://stackoverflow.com/questions/58471615/is-it-possible-to-define-enums-without-values-with-python

Comment (neriat) [reply teitelbot/response/action.py]: nice. will do

Comment (weissgilad) [reply teitelbot/response/action.py]: If it's not too ugly

**Review by weissgilad — APPROVED** [review]
There are a lot of commits. Happy squashing :)

File: teitelbot/response/reply/service/email_generator.py (hunk 5 above)
Comment (neriat) [reply teitelbot/response/reply/service/email_generator.py:6]: new issue #33

**Pull request merged**